core/app/models/spree/app_configuration.rb

```diff
@@ -352,6 +352,17 @@ def default_payment_builder_class
       @default_payment_builder_class ||= Spree::Wallet::DefaultPaymentBuilder
     end
 
+    # Allows providing your own class for canceling payments.
+    #
+    # @!attribute [rw] payment_canceller
+    # @return [Class] a class instance that responds to `cancel!(payment)`
+    attr_writer :payment_canceller
+    def payment_canceller
+      @payment_canceller ||= Spree::Payment::Cancellation.new(
```

Comment:
I just have some doubt about the naming:

I expected this class to be named `Spree::Payment::Canceler` because it is the entity that performs the cancelation. `Spree::Payment::Cancellation` also makes sense but it's more an action, maybe I'd expect the cancelation to happen right when it is initialized, instead this will not performs anything until the `cancel!` method is called. 

As discussed in private with @tvdeyen it is just a personal preference, both makes sense, I just wanted to express mine 😸 .